universal-login-wallet/src/ui/react/Modals/Transfer/TransferDropdownItem.tsx

```diff
@@ -10,15 +11,27 @@ interface DropdownItemProps {
   onClick: (transferCurrency: string) => void;
 }
 
-export const TransferDropdownItem = ({className, name, symbol, balance, onClick}: DropdownItemProps) => (
-  <button onClick={() => onClick(symbol)} className={className || 'currency-accordion-item'}>
-    <div className="currency-accordion-left">
-      <img src={symbol === 'ETH' ? ethIcon : daiIcon} alt="dai" className="currency-accordion-img" />
-      <p className="currency-accordion-name">{name}</p>
-    </div>
-    <div className="currency-accordion-right">
-      <p className="currency-accordion-amount">{balance} {symbol}</p>
-      <p className="currency-accordion-amount-usd">$--,--</p>
-    </div>
-  </button>
-);
+export const TransferDropdownItem = ({className, name, symbol, balance, onClick}: DropdownItemProps) => {
+  const {sdk, walletPresenter} = useServices();
+  const [usdAmount, setUsdAmount] = useState<string>('');
+
+  useEffect(() => {
+    const promise = sdk.subscribeToAggregatedBalance(walletPresenter.getName(), setUsdAmount, symbol);
+    return () => {
+      promise.then((unsubscribe: () => void) => unsubscribe());
+    };
```

Comment:
What do I miss? Why not following:
```ts
useEffect(() => sdk.subscribeToAggregatedBalance(walletPresenter.getName(), setUsdAmount, symbol));
```

Reply:
I guess we need to to unsubscribe when componentWillUnmount which is now a return callback of useEffect, no?

Reply:
Yes, we should unsubscribe when `componentWillUnmount` but for now `sdk.subscribeToAggregatedBalance` return `Promise<unsubscribe>` instead of `unsubscribe` so we need to resolve this promise and call `unscubscribe` function.

Reply:
Use:
```ts 
useAsyncEffect(() => sdk.subscribeToAggregatedBalance(walletPresenter.getName(), setUsdAmount, symbol));
```

```ts
useAsyncEffect(asyncCallback: () => Promise<() => void>) {
  useEffect(() => {
    const promise = asyncCallback();
    return () => {
      promise.then((unsubscribe: () => void) => unsubscribe());
    };
  });
});
```